### ISSUE TITLE
Allow case-insensitive Content-Type

### ISSUE THREAD

**Issue by zanona** [apigee-127/sway#67]
Hi,
I am not sure if this is the expected behaviour but at the moment validation against request presenting headers with `Content-Type` instead of `content-type` fail with the error:

```js
{
  code: 'INVALID_CONTENT_TYPE',
  message: 'Invalid Content-Type (application/octet-stream).  These are supported: application/json'
}
```

even thought `Content-Type` was set to `application/json`
I am wondering if wouldn't it be a good idea to analyse Content-Type by ignoring case?
When using NodeJS' `HTTPClientRequest` Content-Type is lower-case but for many other use cases, such as browser requests, it's would be expected to be in camel case?

I am particularly seeing it very useful to use the `Operation.validateRequest(req)` command with a custom object containing a possible request which hasn't been made up to that point (just for the sake of validation), by providing:

```js
{
    url: '/user',
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    query: { },
    body: { name: 'John Appleseed' }
}
```

We could achieve that by simply allowing this small change at:
https://github.com/apigee-127/sway/blob/master/lib/types/operation.js#L245

```js
helpers.validateContentType(req.headers['Content-Type'] || req.headers['content-type'] || 'application/octet-stream', this.consumes, results);
```

**Comment by whitlockjc** [comment]
Yeah, we should probably not care about it.  Most Node.js frameworks I've used always lowercase them and I think even in the tests that's what I do but I can easily make this work.

**Comment by whitlockjc** [comment]
There was more to it than the suggested fix above.  Long story short, header names are case insensitive and there were a number of places where we had to support this properly.  But it's fixed and will be included in the upcoming `v1.0.0` release.

**Comment by zanona** [comment]
thanks a lot Jeremy.
